ValveAnnulusAnalysis/HeartValveWidgets/ValveSequenceBrowserWidget.py

```diff
@@ -0,0 +1,234 @@
+    self.ui.removeTimePointButton.visible = not enabled
+    self.ui.cardiacCyclePhaseSelector.enabled = not enabled
+    self.updateGUIFromMRML()
+
+  def __init__(self, parent=None): # parent should be a layout
+    self.ui = slicer.util.loadUI(resourcePath("UI/ValveSequenceBrowserWidget.ui"))
+    self.ui.setMRMLScene(slicer.mrmlScene)
+
+    # Just used for keeping track of the observers
+    self._heartValveNode = None
+    self._heartValveNodeObserver = None
+
+    self._valveVolumeBrowserNode = None
+    self._valveVolumeBrowserNodeObserver = None
+
+    self._valveBrowserNode = None
+    self._valveBrowserNodeObserver = None
+
+    self.valveModel = None
+    self.valveBrowser = None
+
+    self._readOnly = False
+
+    self.lastValveBrowserSelectedItemIndex = -1
+
+    self.setup(parent)
+
+  def setup(self, parent):
+    if parent is not None:
+      parent.addWidget(self.ui)
+
+    for cardiacCyclePhaseName in CARDIAC_CYCLE_PHASE_PRESETS.keys():
+      self.ui.cardiacCyclePhaseSelector.addItem(cardiacCyclePhaseName)
+
+    # connections
+    self.ui.addTimePointButton.clicked.connect(self.onAddTimePointButtonClicked)
+    self.ui.removeTimePointButton.clicked.connect(self.onRemoveTimePointButtonClicked)
+    self.ui.cardiacCyclePhaseSelector.connect("currentIndexChanged(int)", self.onCardiacCyclePhaseChanged)
+    self.ui.goToAnalyzedFrameButton.clicked.connect(self.onGoToAnalyzedFrameButtonClicked)
+
+  def _removeHeartValveBrowserNodeObserver(self):
+    if self._valveBrowserNode and self._valveBrowserNodeObserver:
+      self._valveBrowserNode.RemoveObserver(self._valveBrowserNodeObserver)
+      self._valveBrowserNodeObserver = None
+
+  def _removeHeartValveNodeObserver(self):
+    if self._heartValveNode and self._heartValveNodeObserver:
+      self._heartValveNode.RemoveObserver(self._heartValveNodeObserver)
+      self._heartValveNodeObserver = None
+
+  def _removeValveVolumeBrowserObserver(self):
+    if self._valveVolumeBrowserNode and self._valveVolumeBrowserNodeObserver:
+      self._valveVolumeBrowserNode.RemoveObserver(self._valveVolumeBrowserNodeObserver)
+      self._valveVolumeBrowserNodeObserver = None
+
+  def setHeartValveBrowserNode(self, heartValveBrowserNode):
+    self.ui.heartValveBrowserPlayWidget.setMRMLSequenceBrowserNode(heartValveBrowserNode)
+    self.ui.heartValveBrowserSeekWidget.setMRMLSequenceBrowserNode(heartValveBrowserNode)
+
+    self.valveBrowserNode = heartValveBrowserNode
+    self.valveBrowser = HeartValveLib.HeartValves.getValveBrowser(heartValveBrowserNode)
+    self.valveVolumeBrowserNode = self.valveBrowser.volumeSequenceBrowserNode if self.valveBrowser else None
+
+    if self.valveBrowser:
+      valveVolumeNode = self.valveVolumeNode
+      if not valveVolumeNode:
+        self._setValveVolumeToBackgroundVolume()
+
+    self.setHeartValveNode(self.valveBrowser.heartValveNode if self.valveBrowser else None)
+
+  def setHeartValveNode(self, heartValveNode):
```

Comment:
It would simplify usage if there was only one public method for setting the heartValveNode from outside (currently, you could consider using `widget.heartValveNode = ...` or `widget.setHeartvalveNode(...)`). Could the content of this method be moved into the property setter? The same question/suggestion applies to `setHeartValveBrowserNode`.